File: go/sorting-records/sorting_records.go
package main

import (
	"database/sql"
	"fmt"

	_ "modernc.org/sqlite"
)

const file string = "./employees.db"

func main() {
	type Employee struct {
		FirstName string
		LastName  string
		Position  string
		Date      string
	}
	employees := []Employee{}

	db, err := sql.Open("sqlite", file)
	if err != nil {
		fmt.Println(err)
		return
	}
	defer db.Close()

	rows, err := db.Query("SELECT * from employees")
	if err != nil {
		fmt.Println(err)
		return
	}

	for rows.Next() {
		var emp Employee

		rows.Scan(&emp.FirstName, &emp.LastName, &emp.Position, &emp.Date)
		employees = append(employees, emp)
	}

	fmt.Printf("%-19s | %-17s | %s\n", "Name", "Position", "Separation Date")
	fmt.Println("-----------------------------------------------------")
	for _, emp := range employees {
		fmt.Printf("%-19s | %-17s | %s\n",
			emp.FirstName+" "+emp.LastName,
			emp.Position,
			emp.Date,
		)
	}
}

File: python/sorting_records/sorting_records.py
import sqlite3
from typing import Any


# Create the db
with sqlite3.connect("./employees.db") as connection:
    cursor = connection.cursor()

employees: list[dict[str, Any]] = []

# execute query
rows = cursor.execute("SELECT * FROM employees").fetchall()


for name, surname, position, date in rows:
    employees.append(
        {
            "name": name,
            "surname": surname,
            "position": position,
            "date": date,
        }
    )

# sorted(employees, key=lambda employee: employee["surname"])
employees.sort(key=lambda emp: emp["surname"])

print(f"{'Name':20}| {'Position':12}| Separation Date")
print(f"{'-'*20}| {'-'*12}| {'-'*15}")
for emp in employees:
    name = f"{emp["name"]} {emp["surname"]}"
    print(f"{name:20}| {emp["surname"]:12}| {emp["date"] or ''}")
